test/cases_source_filter_type_array_dynamic/1.callbacks.c

```diff
@@ -0,0 +1,12 @@
+#include <metababel/metababel.h>
+
+void btx_push_usr_messages(void *btx_handle, void *usr_data,
+                           btx_source_status_t *status) {
+  long int foo[] = {1, 2, 3};
```

Comment:
```suggestion
int size = 3;
long int *foo = (int*)malloc(size * sizeof(int));
for (int j = 0; j < size; ++j) foo[j] = j + 1;
```

Reply:
Should be dynamic allocation?

Reply:
Now it's ok, the lifetime is good. It will not go after the end of this function. So we are good

Reply:
I think it is worth to do the change to the dynamic allocation. I found we do not free in any place in the source. I think that's why I got this.

Static array, everthing OK !!

```bash
e_array_dynamic: { length = 3, dummy = 12, entries = [ [0] = 1, [1] = 2, [2] = 3 ] }

==53621== 
==53621== HEAP SUMMARY:
==53621==     in use at exit: 23,530 bytes in 14 blocks
==53621==   total heap usage: 1,513 allocs, 1,499 frees, 355,789 bytes allocated
==53621== 
==53621== LEAK SUMMARY:
==53621==    definitely lost: 50 bytes in 1 blocks
==53621==    indirectly lost: 0 bytes in 0 blocks
==53621==      possibly lost: 0 bytes in 0 blocks
==53621==    still reachable: 23,480 bytes in 13 blocks
==53621==         suppressed: 0 bytes in 0 blocks
==53621== Rerun with --leak-check=full to see details of leaked memory
==53621== 
==53621== For lists of detected and suppressed errors, rerun with: -s
==53621== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

**Dynamic Array, I got lost extrabytes. Is this OK?**

```bash
e_array_dynamic: { length = 3, dummy = 12, entries = [ [0] = 1, [1] = 2, [2] = 3 ] }
==66041== 
==66041== HEAP SUMMARY:
==66041==     in use at exit: 23,554 bytes in 15 blocks
==66041==   total heap usage: 1,514 allocs, 1,499 frees, 355,813 bytes allocated
==66041== 
==66041== LEAK SUMMARY:
==66041==    definitely lost: 74 bytes in 2 blocks
==66041==    indirectly lost: 0 bytes in 0 blocks
==66041==      possibly lost: 0 bytes in 0 blocks
==66041==    still reachable: 23,480 bytes in 13 blocks
==66041==         suppressed: 0 bytes in 0 blocks
==66041== Rerun with --leak-check=full to see details of leaked memory
==66041== 
==66041== For lists of detected and suppressed errors, rerun with: -s
==66041== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```